sotodlib/tod_ops/fft_ops.py

```diff
@@ -323,41 +334,201 @@ def calc_wn(aman, pxx=None, freqs=None, low_f=5, high_f=10):
     return wn
 
 
-def noise_model(f, p):
+def noise_model(f, params, **fixed_param):
     """
     Noise model for power spectrum with white noise, and 1/f noise.
+    If any fixed param is handed, that parameter is fixed in the fit.
+    'alpha' or 'wn' can be fixed.
     """
-    fknee, w, alpha = p[0], p[1], p[2]
-    return w * (1 + (fknee / f) ** alpha)
+    if 'alpha' in fixed_param.keys():
+        if len(params)==2:
+            alpha = fixed_param['alpha']
+            fknee, wn = params[0], params[1]
+        else:
+            raise ValueError('The number of fit parameters are invalid.')
+            return
+    elif 'wn' in fixed_param.keys():
+        if len(params)==2:
+            wn = fixed_param['wn']
+            fknee, alpha = params[0], params[1]
+        else:
+            raise ValueError('The number of fit parameters are invalid.')
+            return
+    elif len(fixed_param)==0:
+        if len(params)==3:
+            fknee, wn, alpha = params[0], params[1], params[2]
```

Comment:
Can you modify the order of parameters?
I think the order of [wn, fknee, alpha] is more natural and less confusing.